src/bindings/python/tests/test_inference_engine/test_infer_request.py

```diff
@@ -298,16 +300,29 @@ def test_query_state_write_buffer(device, input_shape, data_type, mode):
             mem_state.state = tensor
 
             res = request.infer({0: np.full(input_shape, 1, dtype=data_type)})
+            k = list(res.keys())[0]
             expected_res = np.full(input_shape, 1 + const_init, dtype=data_type)
         elif mode == "reset_memory_state":
             # reset initial state of ReadValue to zero
             mem_state.reset()
             res = request.infer({0: np.full(input_shape, 1, dtype=data_type)})
-
+            k = list(res.keys())[0]
             # always ones
             expected_res = np.full(input_shape, 1, dtype=data_type)
         else:
             res = request.infer({0: np.full(input_shape, 1, dtype=data_type)})
+            k = list(res.keys())[0]
             expected_res = np.full(input_shape, i, dtype=data_type)
-        assert np.allclose(res[0], expected_res, atol=1e-6), \
+        assert np.allclose(res[k], expected_res, atol=1e-6), \
             "Expected values: {} \n Actual values: {} \n".format(expected_res, res)
+
+
+def test_get_results(device):
+    core = Core()
+    func = core.read_model(test_net_xml, test_net_bin)
+    core.set_config({"PERF_COUNT": "YES"}, device)
```

Comment:
I think it's better to add a test with AsyncInferQueue as `request.results` API is more needed for `async` mode rather than `sync`

Reply:
Added.